go/mysql/binlog_event_common.go

```diff
@@ -119,8 +119,9 @@ func (ev binlogEvent) Length() uint32 {
 }
 
 // NextPosition returns the nextPosition field from the header
-func (ev binlogEvent) NextPosition() uint32 {
-	return binary.LittleEndian.Uint32(ev.Bytes()[13 : 13+4])
+func (ev binlogEvent) NextPosition() uint64 {
+	// Only 4 bytes are used for the next_position field in the header.
+	return uint64(binary.LittleEndian.Uint32(ev.Bytes()[13 : 13+4]))
```

Comment:
@mattlord it should probably happen in another iteration, but could the underlying `uint32` overflow?

Reply:
No, it wouldn't overflow as it's only 4 bytes. I'll change the 13+4 there to 17 though as I'm guessing that's what caught your eye and raised the question. There's no reason not to use a constant literal there anyway. Thanks!

Reply:
@mattlord what caught my eye is we are using binary.LittleEndian.Uint32() and not .Uint64()

The latter returns uint64 whereas now we convert 32 to 64

Reply:
@timvaillancourt This reads 4 bytes, which by definition is `4 * 8 = 32` bits. So the protocol only has 4 bytes for it, so it's always a 32 bit value? 

It then is upcasted to a 64 bit value since other paths can lead to a 64 bit value, but this here seems correct to me? Not sure how anything here can then overflow?

Reply:
Yeah, we're reading 32 bits from the header and then adding 0 bit/ digit padding to the most significant 32 bits of the new 64 bit value

Reply:
> So the protocol only has 4 bytes for it, so it's always a 32 bit value?

@dbussink / @mattlord: ahh thanks, wasn't aware of the protocol details - just blindly suggested using 64-bit all the way down 🙈. LGTM